experimental/table/base/src/main/java/alluxio/experimental/ProtoUtils.java

```diff
@@ -143,4 +148,37 @@ private static Type fromProto(alluxio.grpc.catalog.Type type) {
     return new org.apache.iceberg.Schema(schema.getColsList().stream()
         .map(ProtoUtils::fromProto).collect(Collectors.toList()));
   }
+
+  /**
+   * @param partition the partition proto
+   * @return true if the partition has the hive layout, false otherwise
+   */
+  public static boolean isHiveLayout(Partition partition) {
+    if (!partition.hasLayout()) {
+      return false;
+    }
+    Layout layout = partition.getLayout();
+    // TODO(gpang): use a layout registry
+    return Objects.equals(layout.getLayoutType(), "hive");
+  }
+
+  /**
+   * @param partition the partition proto
+   * @return the hive partition proto
+   */
+  public static PartitionInfo extractHiveLayout(Partition partition)
```

Comment:
Slightly strange to return `PartitionInfo` when the method references `layout`?

Reply:
yeah, this is mostly because we are introducing a concept of a layout. I will refactor protos later, but there will always be an extra level of hierarchy now.